integration-libs/opf/quick-buy/core/services/opf-quick-buy-transaction.service.ts

```diff
@@ -0,0 +1,177 @@
+  protected userAddressService = inject(UserAddressService);
+  protected opfGlobalMessageService = inject(OpfGlobalMessageService);
+
+  getTransactionDeliveryType(): Observable<OpfQuickBuyDeliveryType> {
+    return this.activeCartFacade.hasDeliveryItems().pipe(
+      take(1),
+      map((hasDeliveryItems: boolean) =>
+        hasDeliveryItems
+          ? OpfQuickBuyDeliveryType.SHIPPING
+          : OpfQuickBuyDeliveryType.PICKUP
+      )
+    );
+  }
+
+  getTransactionDeliveryInfo(): Observable<OpfQuickBuyDeliveryInfo> {
+    const deliveryTypeObservable = this.getTransactionDeliveryType().pipe(
+      map((deliveryType) => {
+        return {
+          type: deliveryType,
+        } as OpfQuickBuyDeliveryInfo;
+      })
+    );
+
+    return deliveryTypeObservable.pipe(take(1));
+  }
+
+  getTransactionLocationContext(): Observable<OpfQuickBuyLocation> {
+    return this.routingService.getRouterState().pipe(
+      take(1),
+      map(
+        (routerState) =>
+          routerState?.state?.semanticRoute?.toLocaleUpperCase() as OpfQuickBuyLocation
+      )
+    );
+  }
+
+  getMerchantName(): Observable<string> {
+    return this.baseSiteService.get().pipe(
+      take(1),
+      map((baseSite) => baseSite?.name ?? OPF_QUICK_BUY_DEFAULT_MERCHANT_NAME)
+    );
+  }
+
+  checkStableCart(): Observable<boolean> {
+    return this.activeCartFacade.isStable().pipe(
+      filter((isStable) => !!isStable),
+      take(1)
+    );
+  }
+
+  getSupportedDeliveryModes(): Observable<DeliveryMode[]> {
+    return this.checkoutDeliveryModesFacade.getSupportedDeliveryModes();
+  }
+
+  setDeliveryAddress(address: Address): Observable<string> {
+    this.opfGlobalMessageService.disableGlobalMessage([
+      'addressForm.userAddressAddSuccess',
+    ]);
+    return this.checkoutDeliveryAddressFacade.createAndSetAddress(address).pipe(
+      switchMap(() => this.checkStableCart()),
+      switchMap(() =>
+        this.getDeliveryAddress().pipe(
+          map((addr: Address | undefined) => addr?.id ?? '')
+        )
+      )
+    );
+  }
+
+  setBillingAddress(address: Address): Observable<boolean> {
+    return this.checkoutBillingAddressFacade
+      .setBillingAddress(address)
+      .pipe(switchMap(() => this.checkStableCart()));
+  }
+
+  getDeliveryAddress(): Observable<Address | undefined> {
+    return this.checkoutDeliveryAddressFacade.getDeliveryAddressState().pipe(
+      filter((state: QueryState<Address | undefined>) => !state.loading),
+      take(1),
+      map((state: QueryState<Address | undefined>) => {
+        return state.data;
+      })
+    );
+  }
+
+  getCurrentCart(): Observable<Cart> {
+    return this.activeCartFacade.takeActive();
+  }
+
+  getCurrentCartId(): Observable<string> {
+    return this.activeCartFacade.takeActiveCartId();
+  }
+
+  getCurrentCartTotalPrice(): Observable<number | undefined> {
+    return this.activeCartFacade
+      .getActive()
```

Comment:
getActive() is a stream that never complete, we should you takeActive() or this.getCurrentCart().

Reply:
Thanks, I did a change.